### PR TITLE
Marketplace: Add search header 

### LINKED ISSUE #61863: Marketplace: Add search header
We need to add a search header to the /plugins page. Figma: f84fqtCG4pCnCh2Hq0But5-fi-1164%3A15551

Screen Shot 2022-03-11 at 5 20 02 PM

- [x] Re-use header and search components to add the search.
- [ ] Implement scroll experience. @vinimotaa to confirm if search will stay in fixed breadcrumb header.
- [x] Add "most popular searches". @cpapazoglou Do we have data to use here?

### PR DESCRIPTION
**Pull request by Rolilink**
#### Changes proposed in this Pull Request

* Adds a header to the marketplace.
* Moves the search component from the navigation bar to the header.
* Adds a list of recommended searches.

#### Screenshots
| Mobile | Tablet | Desktop |
|--------|-------|----------|
|Screen Shot 2022-03-23 at 4 20 29 PM|Screen Shot 2022-03-23 at 4 20 09 PM|Screen Shot 2022-03-23 at 4 19 30 PM|

#### Testing instructions

* Navigate into `http://calypso.localhost:3000/plugins`.
* It should render the heading, the search bar and recommended searches.
* Write `seo` on the search bar it should show `seo` related plugins.
* Click the `portfolio` recommended search, it should show `portfolio` related plugins.
* Navigate into `http://calypso.localhost:3000/plugins/:siteSlug`.
* It should render the heading, the search bar and recommended searches.
* Write `seo` on the search bar it should show `seo` related plugins.
* Click the `portfolio` recommended search, it should show `portfolio` related plugins.

Related to #61863

### CODE REVIEW

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 1]
```diff
@@ -354,6 +364,47 @@ const PluginsBrowser = ( {
...
+						className="plugins-browser__recommended-searches-list-item"
+					>
+						{ searchTerm }
+					</a>
+				) ) }
+			</div>
+		</div>
+	);
+};
+
```
Comment (cpapazoglou) [inline client/my-sites/plugins/plugins-browser/index.jsx]: Can we move new components under `plugins` for the `SearchHeader`? This file has already become too big and hard to handle.

File: client/my-sites/plugins/plugins-browser/index.jsx [hunk 2]
```diff
@@ -296,9 +303,6 @@ const PluginsBrowser = ( {
 
 						<UploadPluginButton isMobile={ isMobile } siteSlug={ siteSlug } />
 					</div>
-					<div className="plugins-browser__searchbox">
-						{ <SearchBox isMobile={ isMobile } doSearch={ doSearch } search={ search } /> }
```
Comment (cpapazoglou) [inline client/my-sites/plugins/plugins-browser/index.jsx:300]: Should we also remove the `SearchBox` related code?

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 1 above)
Comment (cpapazoglou) [reply client/my-sites/plugins/plugins-browser/index.jsx]: Bundling their respective CSS and removing it from `client/my-sites/plugins/plugins-browser/style.scss` too.

File: client/components/fixed-navigation-header/index.tsx [hunk 3]
```diff
@@ -54,53 +54,31 @@ const ActionsContainer = styled.div`
...
+	} = props;
+	const { headerRef, actionsRef } = useSticky( contentRef );
 
 	return (
 		<Header id={ id } className={ className } ref={ headerRef }>
 			<Container>
 				<Breadcrumb items={ navigationItems } compact={ compactBreadcrumb } />
-				<ActionsContainer ref={ actionsRef }>{ children }</ActionsContainer>
+				<ActionsContainer ref={ actionsRef }>{ actions }</ActionsContainer>
+				{ children }
```
Comment (cpapazoglou) [inline client/components/fixed-navigation-header/index.tsx]: What's the rationale behind this change? This component was created to have two sections: 
- left - `Breadcrumb`
- right - `ActionsContainer` (as arbitrary children)

Do we really need a third section? Does unwrapping the `children` create any regressions?

File: client/my-sites/plugins/plugins-browser/index.jsx (hunk 1 above)
Comment (Rolilink) [reply client/my-sites/plugins/plugins-browser/index.jsx]: sure, was thinking on doing that before pushing the PR to review

File: client/components/fixed-navigation-header/index.tsx (hunk 3 above)
Comment (Rolilink) [reply client/components/fixed-navigation-header/index.tsx]: @cpapazoglou I did this to be able to accommodate the sticky behavior of the `SearchBox` component, right now the `FixedNavigationHeader` behavior is to render the breadcrumbs and conditionally rendering the action section which is the one where the children's are rendered an example of this behavior can be seen on the `woocommerce-installation` flow.

To be able to stick the search bar without removing the actions at the right I added this changes, but I wonder if there is a better approach which doesn't impact the `FixedNavigationHeader`.

Comment (Rolilink) [reply client/components/fixed-navigation-header/index.tsx]: Also I tested the flow where we use the `FixedNavigationHeader` and no regressions or weird behavior where caused.

File: client/my-sites/plugins/search-box-header/index.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+		<div className="search-box-header">
+			<div className="search-box-header__header">{ title }</div>
+			<div className="search-box-header__search">
+				<SearchBox doSearch={ doSearch } search={ search } />
+			</div>
+			<PopularSearches siteSlug={ siteSlug } searchTerms={ searchTerms } />
+		</div>
+	);
+};
+
```
Comment (lsl) [inline client/my-sites/plugins/search-box-header/index.jsx:67]: Thank you for breaking things down like this. Everything being simple top level components is a nice way to make things re-usable in future.

**Review by lsl — APPROVED** [review]
LGTM

**Review by cpapazoglou — APPROVED** [review]
Thanks for addressing all feedback and also splitting this which makes it far easier to review and merge! LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7205638

Thank you @Rolilink for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7205638).
